vnda/utils/segment.ts

```diff
@@ -0,0 +1,51 @@
+import { getCookies, setCookie } from "std/http/mod.ts";
+import { AppContext } from "../mod.ts";
+
+interface Segment {
+  agent: string;
+}
+
+export const SEGMENT_COOKIE_NAME = "vnda_segment";
+const SEGMENT = Symbol("segment");
+const SIXTYDAYS = new Date(Date.now() + 60 * 24 * 60 * 60 * 1000);
+
+export const getSegmentFromBag = (ctx: AppContext): string =>
+  ctx.bag?.get(SEGMENT);
+export const setSegmentInBag = (ctx: AppContext, segment: string) =>
+  ctx.bag?.set(SEGMENT, segment);
+
+export const parse = (cookie: string) => JSON.parse(atob(cookie));
+
+export const buildSegmentCookie = (req: Request): string | null => {
+  const url = new URL(req.url);
+  const param = url.searchParams.get("agent");
+  if (param) {
+    const partialSegment: string = param;
+    return partialSegment;
+  }
+  return null;
+};
+
+export const getSegmentFromCookie = (
+  req: Request,
+): string | undefined => {
+  const cookies = getCookies(req.headers);
+  const cookie = cookies[SEGMENT_COOKIE_NAME];
+  return cookie;
+};
+
+export const setSegmentCookie = (
+  segment: string,
+  headers: Headers = new Headers(),
+): Headers => {
+  setCookie(headers, {
+    value: segment,
+    name: SEGMENT_COOKIE_NAME,
+    path: "/",
+    secure: true,
+    httpOnly: true,
+    expires: SIXTYDAYS,
```

Comment:
Could you please verify the expected behavior with a representative from Vnda?

My hypothesis is that it might be better a session cookie or a briefly-lasting cookie (perhaps 1 day).

Reply:
Vnda has communicated that it is essential to keep this cookie active for a period of sixty days, a necessity dictated by the utilization of a tool called 'direct' within the Vnda platform.

The client, while currently don't have any purchase plans in this first days, could decide to buy something within this timeframe. In this case I need to create a cart with this agent.